### ISSUE TITLE
Allow the user to supply her own policy

### ISSUE THREAD

**Issue by masomel** [coniks-sys/libmerkleprefixtree-go#6]
As an extension to the current STR implementation, it would be useful to allow the user to supply her own hash scheme, and other user-specific policies.

**Comment by c633** [comment]
The original idea of @masomel  is
> My idea for a developer-supplied policy was really more about allowing them to set the policy fields we give them to their own values, and less about letting them implement new policy types.

Should we rename the `DefaultPolicies` to `ConiksPolicies` to avoid misunderstanding?
/cc @arlolra @Liamsi

**Comment by Liamsi** [comment]
> Should we rename the `DefaultPolicies` to `ConiksPolicies` to avoid misunderstanding?

Makes sense to me.

**Comment by masomel** [comment]
> Should we rename the ``DefaultPolicies`` to ``ConiksPolicies`` to avoid misunderstanding?

Also sounds good to me. My original idea aside, I do also like @c633's original approach of allowing users to define their own ``Policies`` interface since it gives them a lot of flexibility. But the ``ConiksPolicies`` might be sufficient for developers right now, and as we've been discussing in #37, it complicates things when reconstructing server state from the DB. I think we should keep the user-supplied policy interface as an open issue to address in the future.

**Comment by arlolra** [comment]
Can you clarify in the description / title what's meant by "user" here?  The senses that I'm thinking about are "developer", "account", and "contact".  When I originally read this, I thought you meant the account policies (namely, encrypted values in the merkle tree blob).  I suppose the description disambiguates that a bit, but even still.

**Comment by masomel** [comment]
> Can you clarify in the description / title what's meant by "user" here? The senses that I'm thinking about are "developer", "account", and "contact".

Yeah, the title should say "developer". I opened this issue when we were using "developer" and "user" interchangeably, and I never changed it once we disambiguated the two terms.

**Comment by arlolra** [comment]
Thanks!
